### PR TITLE
Made scheme, input_format, and build_directory arguments optional to …

### PR DESCRIPTION
**Pull request by taiheng**
…match slather requirements.

This fixes krausefx/fastlane#650.

### CODE REVIEW

**Comment by KrauseFx** [comment]
Looks good, thanks for the pull request. Can you also update the tests so that they don't fail? You can run them using `rspec`. More information in the [countdown repo](https://github.com/fastlane/countdown).

I can also take over from here if you prefer.

**Comment by taiheng** [comment]
I'll take a look at the tests, sorry about that.

**Comment by KrauseFx** [comment]
Cool, no problem. Let me know if it doesn't work and I should take over migrating the tests.

**Comment by taiheng** [comment]
I've updated the tests to reflect that the only parameter required is the project name, and also changed the name of this parameter to better match the default actions in fastlane from 'proj' to 'xcodeproj'. This parameter name also seems out of place with the other parameters in this action and is the only one that is truncated for some reason.

File: lib/fastlane/actions/slather.rb [hunk 1]
```diff
@@ -49,22 +49,23 @@ def self.available_options
...
                                        env_name: "FL_SLATHER_BUILD_DIRECTORY", # The name of the environment variable
                                        description: "The location of the build output", # a short description of this parameter
-                                       verify_block: proc do |value|
-                                         raise "No Build Directory specified, pass using `build_directory: 'location/of/your/build/output'`".red unless value and !value.empty?
-                                       end),
-          FastlaneCore::ConfigItem.new(key: :proj,
-                                       env_name: "FL_SLATHER_PROJ", # The name of the environment variable
+                                       optional: true
+                                      ),
+          FastlaneCore::ConfigItem.new(key: :xcodeproj,
```
Comment (KrauseFx) [inline lib/fastlane/actions/slather.rb]: Sorry, we can't just change the key of this option, as it would break existing setups. Please revert this change to be `:proj` again.

**Comment by cyupa** [comment]
Any chance you'll merge this, @KrauseFx?

**Comment by taiheng** [comment]
Reverted breaking change and test.

File: lib/fastlane/actions/slather.rb [hunk 2]
```diff
@@ -49,22 +49,23 @@ def self.available_options
...
                                        env_name: "FL_SLATHER_BUILD_DIRECTORY", # The name of the environment variable
                                        description: "The location of the build output", # a short description of this parameter
-                                       verify_block: proc do |value|
-                                         raise "No Build Directory specified, pass using `build_directory: 'location/of/your/build/output'`".red unless value and !value.empty?
-                                       end),
+                                       optional: true
+                                      ),
           FastlaneCore::ConfigItem.new(key: :proj,
-                                       env_name: "FL_SLATHER_PROJ", # The name of the environment variable
+                                       env_name: "FL_SLATHER_PROJECT", # The name of the environment variable
```
Comment (KrauseFx) [inline lib/fastlane/actions/slather.rb]: :cry:

**Comment by KrauseFx** [comment]
Thanks @taiheng
